### ISSUE TITLE
ResultsExpireIn never

### ISSUE THREAD

**Issue by sstarcher** [RichardKnop/machinery#191]
Would you take a PR to update the ResultsExpireIn to support never expiring the results?  Currently the ResultsExpireIn is an int so the longest results can be kept would be 45 days.

**Comment by SurendraPlivo** [comment]
@sstarcher right now its not implemented in the machinery. but we can change the behaviour of ResultsExpireIn to never expire the result. i started working on this. i will be coming back with solution.

**Comment by RichardKnop** [comment]
@sstarcher @SurendraPlivo I agree this is not possible with current implementation. It could be modified to allow this behavior. It's simple with some backends (Redis, MongoDB) which are persistent. It's a bit more tricky with Memcache and also AMQP.

I'd suggest your tasks could save data you need to hold longterm in a database. This can be totally separate from machinery and be part of your application database. But also I don't mind if you want to submit PR to make it possible to store results longterm.
